### PR TITLE
Add a style file for Cryptol

### PR DESCRIPTION
**Pull request by weaversa**
To satisfy #10.

### CODE REVIEW

File: cryptol-style.md [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+
+The `=` should always be surrounded by a single space on either
+side. This aligns value definitions with type definitions (where the `:`
+is also surrounded by a single space on either side). For example,
+
+```haskell
+x : [32]
+x = 10
+y : [4][32]
+y = [1, 2, 3, 4]
```
Comment (AssemblyZig) [inline cryptol-style.md:104]: See above re: alignment

File: cryptol-style.md [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+
+```haskell
+Person =
+    { firstName : String 10
+    , lastName  : String 10
+    , age       : Integer
+    }
+```
+
+Align the elements in a list. For example,
```
Comment (AssemblyZig) [inline cryptol-style.md:117]: "... example," --> "...example:"

File: cryptol-style.md [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+===================
+
+This document was crafted from [tibbe's Haskell style
+guide](https://github.com/tibbe/haskell-style-guide/blob/master/haskell-style.md). This
+guide covers the major areas of formatting and naming of Cryptol
+specifications. That said, the overarching goal of writing a Cryptol
+specification is to make it look as much like the corresponding paper
+document as possible. So, if there is some clear style in the paper
+document, follow that as closely as possible and use this guide to
+fill in any gaps.
```
Comment (AssemblyZig) [inline cryptol-style.md:11]: By "paper document", do you mean "human-readable specification"?

File: cryptol-style.md [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+### End-of-Line Comments
+
+Separate end-of-line comments from the code using 2 spaces.
+
+```haskell
+foo : [32] -> [32]
+foo n = n % p
+  where
+    p = 4294967291  -- Largest 32 bit prime.
+```
```
Comment (AssemblyZig) [inline cryptol-style.md:205]: End of line comments aren't referenced in the Programming Cryptol document (https://cryptol.net/files/ProgrammingCryptol.pdf), and my lame attempt with them was unsuccessful...

File: cryptol-style.md [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+data Parser = Parser
+    !Int         -- Current position
+    !ByteString  -- Remaining input
+
+foo :: Int -> Int
+foo n = salt * 32 + 9
+  where
+    salt = 453645243  -- Magic hash salt.
+```
+
```
Comment (AssemblyZig) [inline cryptol-style.md:290]: This seems mostly redundant...

File: cryptol-style.md [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+### Links
+
+Use in-line links economically. You are encouraged to add links for
+API names. It is not necessary to add links for all API names in a
+Haddock comment. We therefore recommend adding a link to an API name
```
Comment (AssemblyZig) [inline cryptol-style.md:295]: "Haddock comment" ???

File: cryptol-style.md [hunk 7]
```diff
@@ -0,0 +1,405 @@
...
+```
+
+As an alternative to the `UNPACK` pragma, you can put
+
+```haskell
+{-# OPTIONS_GHC -funbox-strict-fields #-}
+```
+
+at the top of the file. Including this flag in the file itself instead
+of e.g. in the Cabal file is preferable as the optimization will be
```
Comment (AssemblyZig) [inline cryptol-style.md:365]: "Cabal file" ???

File: cryptol-style.md [hunk 8]
```diff
@@ -0,0 +1,405 @@
...
+Dealing with laziness
+---------------------
+
+By default, use strict data types and lazy functions.
+
+### Data types
+
+Constructor fields should be strict, unless there's an explicit reason
+to make them lazy. This avoids many common pitfalls caused by too much
+laziness and reduces the number of brain cycles the programmer has to
```
Comment (AssemblyZig) [inline cryptol-style.md:329]: In this context, "strict" and "point-free style" are new terms to me.  Are they important at this level?

**Comment by weaversa** [comment]
@AssemblyZig I think you missed the big STOPPED sign that designated where we last stopped when churning the Haskell style guide into Cryptol.

**Comment by weaversa** [comment]
@jpziegler Ready for review!

**Pull request merged**